src/sqlparser/src/ast/statement.rs

```diff
@@ -831,7 +831,13 @@ impl ParseTo for CreateSinkStatement {
 
         let sink_from = if p.parse_keyword(Keyword::FROM) {
             impl_parse_to!(from_name: ObjectName, p);
-            CreateSink::From(from_name)
+            let sink_from = CreateSink::From(from_name);
+            if p.peek_nth_any_of_keywords(0, &[Keyword::WHERE]) {
+                return Err(ParserError::ParserError(
+                    "Unexpected WHERE keyword after the FROM clause".to_string(),
+                ));
+            }
```

Comment:
This can't cover cases other than `where`. A better way can be check if the next token is `with` and report error with `self.expectd`

Reply:
good idea